### ISSUE TITLE
Tablet getting disconnected on clicking Recommended tab

### ISSUE THREAD

**Issue by miyer03** [learningequality/kolibri#6466]
### Observed behavior
On clicking the recommended tab, the tablet gets disconnected from the server. This also occurs when you watch a video / do some activity and click the back button on the recommended tab

### Expected behavior
The tablet should not get disconnected from the server.

### User-facing consequences
Frequent disconnections lead to a lot of chaos and disengagement in the classroom

### Errors and logs
Attached in the kolibri log file.
[kolibri.txt](https://github.com/learningequality/kolibri/files/4064230/kolibri.txt)

### Steps to reproduce
- Connect a tablet to a server.
- Login to Kolibri as a learner.
- Click on the Recommended tab.
- Once you are running a video/ doing a lesson try clicking on the back arrow.

### Context

 * Kolibri version : 13.0
 * Operating system: Ubuntu 16.04
 * Browser : Chrome

**Comment by miyer03** [comment]
Attached are the nginx logs

[access.log](https://github.com/learningequality/kolibri/files/4069475/access.log)
[kolibri_uwsgi.log](https://github.com/learningequality/kolibri/files/4069476/kolibri_uwsgi.log)
[error.log](https://github.com/learningequality/kolibri/files/4069477/error.log)

**Comment by jonboiser** [comment]
@miyer03 In `kolibri.txt`, the only errors I see are from trying to import content from the "Balbharati Maths G3 to G10" channel. Was the server trying to download content while the class was happening?

Aside from, that, what I see in the `access.log` is this pattern:

```
 "GET /api/content/contentnode_slim/popular/?user_kind=learner&contentCacheKey=1578916195 HTTP/1.1" 200 7896
 "GET /api/content/contentnode_slim/2d2abce95401aa05590b2d9abba39d69/resume/?user_kind=learner&contentCacheKey=1578916195 HTTP/1.1" 200 2198
 "GET /api/content/channel/?available=true&contentCacheKey=1578916195 HTTP/1.1" 200 8056
 "GET /api/content/contentnode_slim/2d2abce95401aa05590b2d9abba39d69/next_steps/?user_kind=learner&contentCacheKey=1578916195 HTTP/1.1" 200 32708
 502 Error here -> "GET /api/content/contentnodeprogress/?ids=[VERY VERY VERY LONG STRING OF NODE IDS] HTTP/1.1" 502 18017
```

Basically, these are the network calls for the recommended page, followed by this very long URL to `/contentnodeprogress` that seems to be crashing. (returns 502 error). This may be why the server is crashing and causing disconnections.

Could you send a screenshot of the what "Recommended" page looks like when the tablet doesn't disconnect? Do they Recommended pages have a lot of resources on them?

**Comment by miyer03** [comment]
@jonboiser screen shot of the recommended tab. When we click on the arrow on the right it shows more content. Otherwise on the page it shows less. Notable thing is its happening only for one ID a3 out of the other ids I created. It tested with 2 to 3 ids.

![Screenshot_20160119-052837](https://user-images.githubusercontent.com/8583367/72783800-453d1200-3c4d-11ea-99cd-4c21efab3557.png)

**Comment by indirectlylit** [comment]
Videos from @miyer03:

* https://drive.google.com/file/d/1pWF28qODtZ5S7Dw8cytPLJjS5D4EQBe2/view
* https://drive.google.com/file/d/1yyTIDWq0U3SzoN9b-CE0G7IBNeAx8aPq/view

**Comment by jredrejo** [comment]
@jonboiser  your comment reminded me of this PR I did almost a year ago: #5404 

Also,  looking at his kolibril.txt log I see this error happening a couple of times aprox at the same hours @miyer03 had the problems.:
```
ERROR 2020-01-15 11:57:06,833 exception Internal Server Error: /api/content/remotechannel/ab9a531a0b284717a3979baabc288543/
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/urllib3/connection.py", line 159, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "/usr/lib/python3/dist-packages/kolibri/dist/urllib3/util/connection.py", line 57, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/lib/python3.5/socket.py", line 732, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Temporary failure in name resolution
```
This should not be related as I can't think of kolibri accessing to remotechannel to get the Recommended tab, unless it was done by another user and caused some problems to the server making it show this message.

**Comment by indirectlylit** [comment]
@miyer03 the error posted above (`Temporary failure in name resolution`) implies a network configuration issue, as though the router thinks it might be online and is trying to resolve a DNS request.

[See google search](https://www.google.com/search?ei=pqkoXs2dA4n3-gSpq5_IAg&q=%22Temporary+failure+in+name+resolution%22&oq=%22Temporary+failure+in+name+resolution%22&gs_l=psy-ab.3..0i67l9j0.3821.4495..4780...0.2..0.97.182.2......0....1..gws-wiz.......0i71j0i7i30.kOjDYZB44kQ&ved=0ahUKEwjNiIi1_5fnAhWJu54KHanVBykQ4dUDCAs&uact=5)

If this is the case, I would recommend trying with a different router configuration.

**Comment by lyw07** [comment]
I'm able to reproduce this issue by going to the `Recommended` tab on our main kolibri demo server. My guess is that the the request URL for `contentnodeprogress` endpoint is too long with the number of node ids.
The error in the console:
<img width="1180" alt="Screen Shot 2020-07-27 at 12 19 41 PM" src="https://user-images.githubusercontent.com/19424916/88583153-b26a9f00-d004-11ea-847d-e46557fad7a5.png">
The actual request URL that caused the error: https://learningequality.slack.com/files/U3QCLGV26/F017S5FH8LD/untitled

Kolibri version is 0.13.3

**Comment by rtibbles** [comment]
This was fixed in #7434
